### DISCUSSION TITLE
Recommended approach for dynamic quotas

### DISCUSSION THREAD

**Discussion by krestjaninoff** [kubernetes-sigs/kueue#2512] [Q&A]
It is quite common for large clusters to deal with temporary unavailable nodes due to hardware issues or other reasons. In this case, kueue's cluster queues have to be updated to reflect the actual amount of available resources. Otherwise, we are facing a possibility of a partially started job (when a subset of pods cannot be scheduled due to a lack of available nodes).

What's the recommended way to address this issue? Should it be a custom solution to monitor node health events and update kueue configuration?

**Comment by tenzen-y** [comment]
As my understanding, when you use the GKE and cluster autoscaler, the ClusterQueue automatically adjusted based on the current available Node by ProvisioningRequest.

Where do you work Kueue? GKE? other Cloud Providers? Or OnPrem?

**Reply by tkatila** [reply]
Piggy packing this. I've been wondering the same thing. For me, the use case would be OnPrem with topology aware scheduling. A group of nodes on a rack with highspeed connectivity between them. A workload would need to fit into the nodes on the rack, and not fit if any of the nodes' resources are unhealthy or temporarily decreased.
